rust/src/lexer.rs

```diff
@@ -0,0 +1,256 @@
+//! Includes functionality for reading MCAP records out of a [`std::io::Read`] implementation.
+use crate::parse::{ParseError, Parser};
+use crate::records::{AttachmentHeader, InvalidOpcode, OpCode};
+use std::error::Error;
+
+pub const MAGIC: [u8; 8] = [0x89, b'M', b'C', b'A', b'P', 0x30, b'\r', b'\n'];
+
+#[derive(Debug)]
+pub enum LexError {
+    IO(std::io::Error),
+    ParseError(ParseError),
```

Comment:
doesn't parsing come after lexing? How can a parse error become a lex error?

Reply:
Yeah, maybe I should differentiate somehow. Here the ParseError occurs if we fail to parse a length field (needed for lexing).